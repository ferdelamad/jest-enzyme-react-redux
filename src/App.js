import React, { Component } from "react";
import logo from "./logo.svg";
import "./App.css";

const Title = ({ text }) => <div>{text}</div>;
class App extends Component {
  state = {
    on: false,
    input: ""
  };
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <p>
            Edit <code>src/App.js</code> and save to reload.
          </p>
          <Title text="I am title" />
          <a
            className="App-link"
            href="https://reactjs.org"
            target="_blank"
            rel="noopener noreferrer"
          >
            Learn React
          </a>
          <ul className="tyler">
            <li>test</li>
            <li>test</li>
            <li>test</li>
          </ul>
        </header>
        <div>
          <p className="button-state">{this.state.on ? "Yes!" : "No!"}</p>
          <button onClick={() => this.setState({ on: true })}>Click</button>
          <h2>{this.state.input}</h2>
          <input
            onChange={e => this.setState({ input: e.currentTarget.value })}
            type="text"
          />
        </div>
      </div>
    );
  }
}

export const Link = props => <a href={props.address}>Click</a>;

export default App;
