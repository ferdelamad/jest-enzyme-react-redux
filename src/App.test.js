import React from "react";
import ReactDOM from "react-dom";
import App, { Link } from "./App";

import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";

configure({ adapter: new Adapter() });

describe("<App />", () => {
  it("should contain 1 p element", () => {
    const wrapper = shallow(<App />);
    //shallow(<App />, { context: {}, disableLifecycleMethods: bool})
    expect(wrapper.find("p").length).toBe(1);
  });

  it("should contain className of App", () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find(".App").exists()).toBe(true);
  });

  // it("should contain a ul with 3 li", () => {
  //   const wrapper = shallow(<App />);
  //   expect(wrapper.find("ul").children.length).toBe(3);
  // });

  it("ul should have a className of tyler", () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find("ul").hasClass("tyler")).toBe(true);
  });

  //find by prop
  // it("find element by prop", () => {
  //   const wrapper = shallow(<App />);
  //   expect(wrapper.find('[text="I am title"]').text()).toBe("I am title");
  // });

  //snapshot press U to re-run
  it("matches the snapshot", () => {
    const tree = shallow(<App />);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it("on button click changes p text", () => {
    const wrapper = shallow(<App />);
    const button = wrapper.find("button");
    expect(wrapper.find(".button-state").text()).toBe("No!");
    button.simulate("click");
    expect(wrapper.find(".button-state").text()).toBe("Yes!");
  });

  it("on input change, title changes text", () => {
    const wrapper = shallow(<App />);
    const input = wrapper.find("input");
    expect(wrapper.find("h2").text()).toBe("");
    input.simulate("change", { currentTarget: { value: "Tyler" } });
    expect(wrapper.find("h2").text()).toBe("Tyler");
  });
});

describe("<Link />", () => {
  it("link component accepts address prop", () => {
    const wrapper = shallow(<Link address="www.google.com" />);
    expect(wrapper.instance().props.address).toBe("www.google.com");
  });

  it("a tag node renders href correctly", () => {
    const wrapper = shallow(<Link address="www.google.com" />);

    expect(wrapper.props().href).toBe("www.google.com");
  });
});
