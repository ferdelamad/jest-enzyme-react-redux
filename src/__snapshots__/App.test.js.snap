// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches the snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <p>
      Edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
    <Title
      text="I am title"
    />
    <a
      className="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
    <ul
      className="tyler"
    >
      <li>
        test
      </li>
      <li>
        test
      </li>
      <li>
        test
      </li>
    </ul>
  </header>
  <div>
    <p
      className="button-state"
    >
      No!
    </p>
    <button
      onClick={[Function]}
    >
      Click
    </button>
    <h2 />
    <input
      onChange={[Function]}
      type="text"
    />
  </div>
</div>
`;
